### PR TITLE
Styles cleanup

### PR DESCRIPTION
**Pull request by cchaos**
Mainly reducing the amount of redundant styles.

@lukeelmers

### CODE REVIEW

File: src/components/table/_mixins.scss [hunk 1]
```diff
@@ -6,7 +6,7 @@
 
 @mixin euiTableCellCheckbox {
   @include euiTableCell;
-  border-top: none;
+  // border-top: none;
```
Comment (lukeelmers) [inline src/components/table/_mixins.scss]: Was this intentional? I assume so, just double checking

File: src/components/table/table_footer.js [hunk 2]
```diff
@@ -4,7 +4,7 @@ import PropTypes from 'prop-types';
 export const EuiTableFooter = ({ children, className, ...rest }) => {
   return (
     <tfoot className={className} {...rest}>
-      <tr className="euiTableFooterRow">{children}</tr>
+      <tr>{children}</tr>
```
Comment (lukeelmers) [inline src/components/table/table_footer.js:7]: Just a note that snapshot tests will likely need to be updated as a result of this (I'm happy to do this after we merge)

File: src/components/table/_mixins.scss (hunk 1 above)
Comment (cchaos) [reply src/components/table/_mixins.scss]: Yeah, that line should actually be deleted entirely.

File: src/components/table/table_footer.js (hunk 2 above)
Comment (cchaos) [reply src/components/table/table_footer.js:7]: I ran the updater, but nothing updated.

**Pull request merged**